packages/rum-core/test/dom.ts

```diff
@@ -1,10 +1,18 @@
 import { arrayFrom } from '@datadog/browser-core'
 import { registerCleanupTask } from '@datadog/browser-core/test'
 
-export function append<E extends Element | Text = Element>(
-  html: string,
-  container: Element | ShadowRoot = document.body
-): E {
+export function appendText(text: string, container: Element | ShadowRoot = document.body): Text {
+  const textNode = document.createTextNode(text)
+  container.appendChild(textNode)
+
+  registerCleanupTask(() => {
+    container.removeChild(textNode)
+  })
+
+  return textNode
+}
+
+export function appendElement(html: string, container: Element | ShadowRoot = document.body): HTMLElement {
```

Comment:
Because it uses `tmp.childNotes[0]`, this function can still return a `Text` node. Maybe use `tmp.children[0]` like before? (Note: I noticed some tests where `appendElement` will need to be changed to `appendText`)

Reply:
Yes good idea